app/src/main/kotlin/com/wire/android/util/logging/LogFileWriterV2Impl.kt

```diff
@@ -0,0 +1,374 @@
+     * ```
+     */
+    override suspend fun start() {
+        appLogger.i("KaliumFileWritter.start called")
+        val isWriting = writingJob?.isActive ?: false
+        if (isWriting) {
+            appLogger.d("KaliumFileWriter.init called but job was already active. Ignoring call")
+            return
+        }
+        ensureLogDirectoryAndFileExistence()
+        val waitInitializationJob = Job()
+
+        writingJob = fileWriterCoroutineScope.launch {
+            observeLogCatWritingToLoggingFile().catch {
+                appLogger.e("Write to file failed :$it", it)
+            }.onEach {
+                waitInitializationJob.complete()
+            }.filter {
+                it > config.maxFileSize
+            }.collect {
+                ensureActive()
+                // Flush buffer before compression
+                bufferMutex.withLock {
+                    flushBuffer()
+                }
+                launch { compressAsync() }
+                clearActiveLoggingFileContent()
+                deleteOldCompressedFiles()
+            }
+        }
+
+        // Start periodic flush job
+            flushJob = fileWriterCoroutineScope.launch {
+                while (isActive) {
+                    delay(config.flushIntervalMs)
+                    try {
+                        withTimeout(config.bufferLockTimeoutMs) {
+                            bufferMutex.withLock {
+                                if (logBuffer.isNotEmpty()) {
+                                    flushBuffer()
+                                    lastFlushTime = System.currentTimeMillis()
+                                }
+                            }
+                        }
+                    } catch (e: TimeoutCancellationException) {
+                        appLogger.w("Periodic flush timed out, buffer may be locked by another operation")
+                    } catch (e: Exception) {
+                        appLogger.e("Error during periodic flush", e)
+                    }
+                }
+            }
+
+        appLogger.i("KaliumFileWritter.start: Starting log collection.")
+        waitInitializationJob.join()
+    }
+
+    /**
+     * Observes logcat text, writing to the [activeLoggingFile] as it reads.
+     * @return A Flow that tells the current length, in bytes, of the log file.
+     */
+    private fun CoroutineScope.observeLogCatWritingToLoggingFile(): Flow<Long> = flow<Long> {
+        Runtime.getRuntime().exec("logcat -c")
+        logcatProcess = Runtime.getRuntime().exec("logcat")
+
+        val reader = logcatProcess!!.inputStream.bufferedReader()
+
+        appLogger.i("Starting to write log files, grabbing from logcat")
+        while (isActive) {
+            val text = reader.readLine()
+            if (!text.isNullOrBlank()) {
+                val fileSize = writeLineToFile(text)
+                emit(fileSize)
+            }
+        }
+        reader.close()
+        stopLogcatProcess()
```

Comment:
```suggestion
        logcatProcess?.inputStream?.bufferedReader().use {
            appLogger.i("Starting to write log files, grabbing from logcat")
            while (isActive) {
                val text = readlnOrNull()
                if (!text.isNullOrBlank()) {
                    val fileSize = writeLineToFile(text)
                    emit(fileSize)
                }
            }
            stopLogcatProcess()
        }
```

Question: Can this make it better for resource management and null safety ?

Reply:
for some reason adding this change caused the sync worker to be canceled.